hrim.py

```diff
@@ -0,0 +1,98 @@
+import sys
+sys.path.insert(0, 'scripts')
+import printing
+import parsing
+import compiling
+import os
+from classes import *
+
+def main(args):
+	if args[1] == "show":
+		extend = False
+		# if topic/properties not defined on the parsed file should show their full content or not when printing
+		if len(args) == 4:
+			if args[3] == "1":
+				extend = True
+		# check for file input
+		if len(args) < 3:
+			print "Please input file to parse"
+			sys.exit()
+		module = parsing.main(args[2])
+		printing.main(module, extend)
+	elif args[1] == "generate":
+		extend = False
+		# check for file input
+		if len(args) < 3:
+			print "Please input file to parse"
+			sys.exit()
+		if args[2] == "all":
+			path = os.getcwd()
+			for file in [
```

Comment:
Probably this way of adding folder is very handmade. Can you do something similar but adding all the folder below a directory.

Reply:
https://github.com/erlerobot/HRIM/pull/16#discussion_r240161504